Makefile.toml

```diff
@@ -286,6 +286,52 @@ category = "ICU4X WASM"
 command = "rm"
 args = ["-rf", "wasmpkg"]
 
+### VALGRIND TASKS ###
+
+[tasks.valgrind]
+description = "Build ICU4X example files with default features and run through Valgrind"
+category = "ICU4X Valgrind"
+script_runner = "@duckscript"
+script = '''
+exit_on_error true
+
+valgrind = which valgrind
+assert ${valgrind} "Could not find 'valgrind' in path.\n*** Please run 'apt-get install valgrind' ***"
+
+mkdir benchmarks
+mkdir benchmarks/valgrind
+
+output = exec cargo +nightly-2021-02-28 build --examples --message-format=json --features icu_benchmark_macros/rust_global_allocator --profile bench -Z unstable-options
+if ${output.code}
+    trigger_error "Build failed! To debug, build examples with `--features icu_benchmark_macros/rust_global_allocator`"
+end
+trimmed_stdout = trim ${output.stdout}
+json_messages = split ${trimmed_stdout} "\n"
+for json_message in ${json_messages}
+    json_obj = json_parse ${json_message}
+
+    is_compiler_artifact = eq ${json_obj.reason} "compiler-artifact"
+    is_example = eq ${json_obj.target.kind[0]} "example"
+    if ${is_compiler_artifact} and ${is_example}
```

Comment:
Suggestion (optional): I would appreciate a few comments explaining what's going on here in general.

Reply:
Added comments.